yb-voyager/cmd/importDataFileTaskImporter.go

```diff
@@ -41,6 +41,18 @@ var (
 	MAX_SLEEP_SECOND     = 60
 )
 
+// BatchProducer defines the interface for producing batches from a file.
+type BatchProducer interface {
+	// Done returns true if all batches have been produced.
+	Done() bool
+
+	// NextBatch returns the next batch to be processed, or an error if no more batches are available.
+	NextBatch() (*Batch, error)
+
+	// Close cleans up any resources used by the batch producer.
+	Close()
+}
+
```

Comment:
I think we can have this as `FileBatchProducer` and keep it in a separate file `importDataFileBatchProducer`

Reply:
renaming to "FileBatchProducer" makes sense